src/mapping/NearestNeighborGradientMapping.cpp

```diff
@@ -20,7 +20,7 @@ NearestNeighborGradientMapping::NearestNeighborGradientMapping(
     int        dimensions)
     : NearestNeighborBaseMapping(constraint, dimensions, true, "NearestNeighborGradientMapping", "nng")
 {
-  PRECICE_CHECK(!hasConstraint(CONSERVATIVE), "Nearest-neighbor-gradient mapping is not implemented using a \"conservative\" constraint. Please select constraint=\" consistent\" or a different mapping method.");
+  PRECICE_CHECK(!hasConstraint(CONSERVATIVE), "Nearest-neighbor-gradient mapping is not implemented using a \"conservative\" constraint. Please select constraint=\" consistent\" or a different mapping method. Note that the scaled-consistent method reverts to the method implemented for all mappings but hasn't been specifically tested with nearest-neighbor-gradient.");
```

Comment:
I understand what you want to say, but the user certainly won't, as he or she is not familiar with the implementation details. Also, this warning does not show up in case the user selects `constraint=scaled-consistent`. What we need is something like `PRECICE_WARN(!hasConstraint(SCALEDCONSISTENT), ...` or `PRECICE_CHECK(...` telling the user that the scaled-consistent variant of the NNG mapping has not been tested.